### DISCUSSION TITLE
Improvement: export `ThemeComponent` interface from `MantineTheme.d.ts`

### DISCUSSION THREAD

**Discussion by JustinMartinDev** [mantinedev/mantine#4572] [Feature requests]
## Proposal
Hello, 

I purpose to export the `ThemeComponent` interface contained in `MantineTheme.d.ts` of `@mantine/styles`.

## Why

All types or interfaces in this file are exported instead of `ThemeComponent`. If we want to use `ThemeComponent` we have to define it using this code :

```typescrypt
import { MantineTheme } from "@mantine/core";

type MantineThemeComponents = MantineTheme["components"];
export type ThemeComponent = MantineThemeComponents[keyof MantineThemeComponents];
```

**Comment by rtivital** [comment]
Why can't you use `MantineTheme["components"][string]`?
